Alpha 16 Groundwork

Maxwells-Dragon opened the pull request:
I've started working on updates towards having CCL work in A16.

Right now I can only claim that it compiles and loads (and hopefully does so on other systems), and that I've taken a stab at moving updates from the Core assemblies into the CCL detours.

Hopefully this can save others from spending quite so much time on the more tedious parts of getting the mod up-to-date.

Commit:
Part 1 of trying to make this compile.

Commit:
Alerts now limitted to player home maps.

Gradually slogging through everything that broke in 16.

Not testing anything yet, I just want this to compile.

Commit:
Part 3 of updating the code to compile in V16.  Down to a mere 129 compile errors.

Most of these changes have to do with the fact that the game now uses multiple maps at once, so Find.Map has been replaced with Find.Maps, and many methods now require a map parameter.

I've made my best educated guesses as to how to handle each error this introduces.

Additionally, there have been a number of type and member renames.

I've also added a few "TODOs" for things I'm especially unsure of how to resolve.

Commit:
Merging in the changes that were made since the version I stupidly forked from.

Derp.

(I'm new pretty new to working with github projects)

Commit:
Part 3 of updating the code to compile in V16.

Maxwells-Dragon committed:
Bad find/replace

Commit:
Ok, it finally compiles with the A16 binaries.  There's a still a ton to do, but it's a start.

Commit:
Merge branch 'development' of https://github.com/NuOfBelthasar/CommunityCoreLibrary

Commit:
Importing A16 changes in Assembly-CSharp to the detour code.

Doesn't work yet.  Still debugging.

Commit:
[fixes]

+ detour method parameter type agreement
+ designationCategory null refs

Commit:
[adds]

+ new ToggleSettings
+ handles worldView ToggleSettings.

Now able to create new colonies.

Commit:
[fixes]

+ _CanIngestForJoy was misnamed

Commit:
[fixes]
+ Quickstart

[changes]
+ all tabs replaced with 4 spaces
+ SaveQuitToMain now matches A16 logic
+ Reformatted MainMenuDrawer with only minor logic changes (as part of checking for relevant changes in the A16 Core assemblies, and as part of hunting for bug #174)

Commit:
[fixes]

+ QuitToOS was quiting to main menu

[changes]

+ temporarily disabling push of CCLModTweaks assemblies because of #4
+ switched CCLModTweaks to default profile (was there a reason to use the Client one?  the DLL is tiny)

Commit:
[fixes]

+ a few null refs and missing default constructors that were preventing save files loads

Commit:
[fixes]

+ now reading maps correctly from XML in the Game detour (resolves #10)

Karel-Kroeze commented:
Thank you very much for your hard work!

I've created an alpha16 branch, if you retarget the PR to that I'll accept it - we can then work from there. 

I would also like to invite you to join us on the RimWorld modding discord, as there's some decisions that need to be taken for CCL, and as of now you're the most active developer... Please find Skullywag on the forums and ask him for an invite to the discord (+the CCL channel), tell him Fluffy sent you.

Skullywag commented:
He is already on the Discord and ive now added him to the communitycode room, lets see if we cant get everyone together and sort out a plan of action for how to handle all this.

Commit:
[fixes]

+ minimap no longer rendered in world-view
+ some alert logic was wrong

The alerts aren't showing up when I would expect them to.  Will continue debugging.
